quantum-adapters/kiwoom-adapter/src/kiwoom_api/analysis/api/analysis_router.py

```
@@ -0,0 +1,354 @@
+        
+        # 필요한 일수만큼 자르기
+        chart_data = chart_data_raw[:days] if len(chart_data_raw) > days else chart_data_raw
+        
+        return chart_data
+        
+    except Exception as e:
+        logger.error(f"차트 데이터 조회 실패 ({stock_code}): {e}")
+        raise HTTPException(
+            status_code=500,
+            detail=f"차트 데이터 조회 실패: {str(e)}"
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)